net/http/civetweb/civetweb.c

```diff
@@ -14305,14 +14305,14 @@ set_ssl_option(struct mg_context *ctx)
 	SSL_CTX_set_ecdh_auto(ctx->ssl_ctx, 1);
 #endif /* NO_SSL_DL */
 
-#ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wincompatible-pointer-types"
-#endif
-#if defined(__GNUC__) || defined(__MINGW32__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
-#endif
+//#ifdef __clang__
+//#pragma clang diagnostic push
+//#pragma clang diagnostic ignored "-Wincompatible-pointer-types"
+//#endif
+//#if defined(__GNUC__) || defined(__MINGW32__)
+//#pragma GCC diagnostic push
+//#pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
+//#endif
```

Comment:
Please avoid adding dead code. Can this not be removed? We have the git log to resurrect it if needed.

Reply:
It is more about syncing with original civetweb code. 
I change constant cast at one place and comment-out warning suppression.
Later it will be easier for me merge changes once again from civetweb.

Reply:
It will be different either way, I don't see why this is better than just removing the bad code (possibly even in civetweb itself, since it comes from there). I don't like how ROOT is often inheriting warnings/problems from external projects. Maybe civetweb should be made external, but with a builtin to handle when it's not there, like openssl, for example? What do you think, @Axel-Naumann?

Reply:
civetweb is full http server-side code.
It has minimal deviations from original repository, therefore it is difficult to use it as pure externals.

There is very important feature missing (dynamic number of used threads). I plan to implement it in our copy of the code and then try to submit PR to original repository. 

If you do not like comments - I can remove them.